mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -145,17 +143,22 @@ object CaseKeywordCompletion:
       )
     else
       // Step 1: walk through scope members.
+      def isValid(sym: Symbol) = !parents.isParent(sym)
+        && (sym.is(Case) || sym.is(Flags.Module) || sym.isClass)
+        && parents.isSubClass(sym, false)
       indexedContext.scopeSymbols
-        .foreach(s => visit(s.info.dealias.typeSymbol, s.decodedName, Nil))
+        .map(_.info.dealias.typeSymbol)
+        .filter(isValid(_))
+        .foreach(s => visit(s, s.decodedName, Nil))
 
       // Step 2: walk through known subclasses of sealed types.
       MetalsSealedDesc.sealedStrictDescendants(selectorSym).foreach { sym =>
         val autoImport = autoImportsGen.forSymbol(sym)
         autoImport match
           case Some(value) =>
-            visit(sym.info.dealias.typeSymbol, sym.decodedName, value)
```

Comment:
sealedStrictDescendants are mostly already types, so this is not needed. Also, in 
```scala
enum Color:
  case Red, Blue, Green
```
symbol `Red` is a value of type `Color` so we get incorrect completions (single `Color` completion instead of 3 completions for each color